backend/selectgen.ml

```diff
@@ -1637,9 +1521,7 @@ method emit_tail (env:environment) exp =
      end
   | Cphantom_let (_var, _defining_expr, body) ->
       self#emit_tail env body
-  | Cop((Capply(ty, ((Rc_close_at_apply | Rc_normal) as pos))) as op,
-        args, dbg)
-       when self#tail_call_possible env pos ->
```

Comment:
The test `backtrace/names` fails because we ignore the position here: `f () [@nontail]` is translated to a tail call, disappearing from the backtrace